### DISCUSSION TITLE
poolsolutions

### DISCUSSION THREAD

**Discussion by xiaobailk** [yalmip/YALMIP#1238] [General discussion] [answered]
Hello sir. 
I am a beginner. I solve the model through MATLAB+yalpip+Gurobi. When I set the PoolSolutions attribute to generate a specific number of solutions, and then output each solution through the SolutionNumber attribute, I find that each solution is the same. I am sure that the model has different solutions. This is my output code. How to output them all? Thank u.
`for i = 1:5
    ops = sdpsettings('solver','gurobi','verbos',1,'gurobi.PoolSolutions',10,...
    'gurobi.PoolGap',0,'gurobi.PoolSearchMode',2,'gurobi.SolutionNumber',i);    
    result = optimize(cons,y,ops);
    if result.problem== 0 
        value(x)
        value(y)
    else
        disp('solution wrong');
    end
end`

**Comment by johanlofberg** [comment, marked as answer]
The solutionnumber attribute is connected to some internal machinery connected to post-solve queries on solutions which YALMIP does not use

When using pools, you catch the pool and then do what ever you want with it

```

ops = sdpsettings('savesolveroutput',1,...
result = optimize(...
result.solveroutput.result.pool

ans = 

  1×3 struct array with fields:

    objval
    xn

```

**Reply by xiaobailk** [reply]
Dear Professor Löfberg, Thank you for your answer. I have got the result I wanted using the method you said. Thank you again.
